ui/src/app/views/workspace-edit/component.ts

```diff
@@ -39,13 +39,17 @@ export class WorkspaceEditComponent implements OnInit {
         aggregateAnalysis: false,
         ancestry: false,
         commercialPurpose: false,
-        population: false
+        population: false,
+        reviewRequested: false
       }};
   }
 
   addWorkspace(): void {
     if (!this.buttonClicked) {
       this.buttonClicked = true;
+      if (this.workspace.researchPurpose.reviewRequested) {
+        this.workspace.researchPurpose.timeRequested = new Date().getTime();
```

Comment:
I think this linkage between the boolean and the timestamps should live as far down in the backend as possible. My goal is to avoid any case where they could be set incorrectly / inconsistently; or where you have to know to do it right even though there's only one way to do it right. So the result here would be that you'd just set the boolean, and send that to the backend, and the backend would create the timestamp / ensure consistency.

Brainstorming solutions:
 * Workspaces db object could auto-set related properties when you call setReviewRequested(true).
 * Workspaces db object could not ever actually store the boolean, but use a @Formula (https://gokhan.ozar.net/hibernate-derived-properties-formula-annotation/), not sure if there are any performance concerns.
 * WorkspacesDao manages this, keeps the model dumb. Choke point could be updateFromSwaggerModel or setReviewRequested or something.

Reply:
I like the `setReviewRequested(true)` model. there's a bit more logic that needs to go in to that, but I think that's fine. (To ensure that the value is set once and only once, and then stays the same after that.)